### PR TITLE
DOC: remove performance claim from searchsorted()

### LINKED ISSUE #13566: searchsorted() docs misleading on relative performance
> This function is a faster version of the builtin python bisect.bisect_left (side='left') and bisect.bisect_right (side='right') functions

No, it's not faster for basic cases.

### Reproducing code example:
Python bisect is 3x faster than searchsorted:
```python
from bisect import bisect_right
from numpy import searchsorted

l = list(range(1024 * 8))
m = int(len(l)/3)
%timeit bisect_right(l, m)

a = np.asarray(l)
%timeit searchsorted(a, m)
```
```
444 ns ± 14.7 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
1.57 µs ± 13.7 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

For reference, numba's implementation of `searchsorted()` is a little faster than bisect:
```python
from numba import njit

@njit
def searchsorted_jit(_a, _v):
    return searchsorted(_a, _v)

searchsorted_jit(a, m)
%timeit searchsorted_jit(a, m)
```
```
391 ns ± 2.93 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

### Numpy/Python version information:
```
1.16.2 3.7.3 (default, Mar 27 2019, 09:23:39)
[Clang 10.0.0 (clang-1000.11.45.5)]
```

### PR DESCRIPTION
**Pull request by belm0**
Closes #13566

### CODE REVIEW

**Comment by bashtage** [comment]
It would be better to provide some general guidance, i.e., it is faster for large arrays or when searching for multiple values.

**Comment by bashtage** [comment]
Think about what help message would have been the most useful to you a few days ago before you spent a few hours benchmarking both paths.

**Comment by juliantaylor** [comment]
note that "faster only for large arrays or vector <input>" applies to pretty much all functions in numpy that have equivalents in python.
So I don't think it is not necessary to add it to every single one. Though it doesn't harm either.

**Comment by charris** [comment]
I'd leave the claim out, it serves no purpose. The important distinction is that `searchsorted` searches arrays, uses arrays of items to search for, and returns an array of indexes that can be used for indexing. In other words, it is nothing like `bisect.bisect_{left, right}` except for the search algorithm at the core. Note that the python version treats arrays as sequences. If you simply want to search a sequence, use the python version.

File: numpy/core/fromnumeric.py [hunk 1]
```diff
@@ -1279,7 +1279,7 @@ def searchsorted(a, v, side='left', sorter=None):
     As of NumPy 1.4.0 `searchsorted` works with real/complex arrays containing
     `nan` values. The enhanced sort order is documented in `sort`.
 
-    This function is a faster version of the builtin python `bisect.bisect_left`
+    This function is a version of the builtin python `bisect.bisect_left`
```
Comment (mattip) [inline numpy/core/fromnumeric.py]: This function uses the same algorithm as the builtin python ...

**Comment by belm0** [comment]
I changed the wording according to @mattip's suggestion.

**Comment by mattip** [comment]
Let's go with that, the wording is still a bit clunky but makes no performance claims.

**Pull request merged**